src/main/java/com/datadoghq/sketch/ddsketch/store/PaginatedStore.java

```diff
@@ -0,0 +1,387 @@
+package com.datadoghq.sketch.ddsketch.store;
+
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+/**
+ * This is an unbounded store which allocates storage for counts
+ * in aligned pages stored in an array at offsets modulo the page
+ * size. This means that if a distribution has several modes, the
+ * cost of the storage for the space in between the modes is that
+ * of a null pointer per page, requiring 4-8 bytes (depending on
+ * CompressedOops, whether ZGC is used, and heap size) for each
+ * 2KB page.
+ *
+ * On the contrary, if the data is uniformly distributed filling
+ * each page in a range [N, N + K * PAGE_SIZE) this store will
+ * require K * 20B <strong>extra</strong> space over
+ * {@code UnboundedSizeDenseStore}, because of the 20B metadata
+ * overhead of the array headers for each page.
+ *
+ */
+public final class PaginatedStore implements Store {
+
+    private static final int GROWTH = 8;
+    private static final int PAGE_SIZE = 128;
+    private static final int PAGE_MASK = PAGE_SIZE - 1;
+    private static final int PAGE_SHIFT = Integer.bitCount(PAGE_MASK);
+
+    private double[][] pages = null;
+    private int minPageIndex;
+
+    public PaginatedStore() {
+        this(Integer.MAX_VALUE);
+    }
+
+    PaginatedStore(int minPageIndex) {
+        this.minPageIndex = minPageIndex;
+    }
+
+    PaginatedStore(PaginatedStore store) {
+        this(store.minPageIndex);
+        this.pages = deepCopy(store.pages);
+    }
+
+    @Override
+    public boolean isEmpty() {
+        // won't initialise any pages unless until a value is added,
```

Comment:
Making extra sure with _"unless until"_ 😉

Reply:
I normally say "_until or unless_" but I'll fix this